lib/routes/objects.js

```diff
@@ -386,8 +386,23 @@ module.exports =
             );
           }
 
-          const { latestVersion } = obj;
-          await fastify.putObject({ name, env, headVersion, latestVersion });
+          const {
+            headTimestamp: prevTimestamp,
+            headVersion: prevHeadVersion
+          } = obj;
+
+          if (headVersion === prevHeadVersion) {
+            throw fastify.httpErrors.badRequest(
```

Comment:
This seems more like a 409 error (Conflict) as it's not really a bad request per se, just that the current state of the items makes the request not fulfillable.